iothub/device/src/Serialization/DefaultPayloadConvention.cs

```diff
@@ -78,7 +78,17 @@ internal T GetObject<T>(Stream streamToConvert)
         [SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Method must be instance to benefit from s_settings.")]
         internal T GetObject<T>(string jsonObjectAsText)
         {
-            return JsonConvert.DeserializeObject<T>(jsonObjectAsText);
+            T deserialized;
+
+            try
```

Comment:
I'm a bit out of the loop on this one, so can you explain why we need to have two code paths here? I would expect a deserialize method to only handle one format of string here, not two.

Reply:
Sure - there's an inconsistency when setting direct method payloads from the SDK vs setting them from the portal or Azure CLI. In the latter case, they're a layer of encoding behind the SDK set payloads, the method here tries to deserialize and performs that "extra layer" of encoding if it cannot be. This lack of consistency is partially my fault, since there aren't tests for when the payload is set through the portal or through AZ CLI and this bug didn't come up in the initial testing for the feature that made payloads into byte[]. Rectifying this would be a good idea in the future for SDK work, but it seemed like getting the fix out was the priority.

Reply:
Gotcha, thanks for the context